File: week6/problemset6/cash.py
from cs50 import get_float

# input amount of changed owed
cents = get_float("Change owed: ")
coins = 0

# reprompt if input is invalid
while cents < 0:
    cents = get_float("Changed owed: ")

# convert value to whole number
cents = cents*100

# calculate quarters & remainder
coins += cents//25
cents %= 25
# DEBUG: print("Quarters:", coins)

# calculate dimes and remainder
coins += cents//10
cents %= 10
# DEBUG: print("total:", coins)

# calculate nickels and remainder
coins += cents//5
cents %= 5
# DEBUG: print("total", coins)

# rest added as cents
coins += cents

# print coin total
print(int(coins))
print()

File: week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int cents = 0;
int main(void)
{
    // Prompt the user for change owed, in cents
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Create amount of coins variable
    int coins = 0;

    // Calculate how many quarters you need, and remainder of cents left
    coins += cents / 25;
    cents %= 25;

    // Calculate how many dimes you need, and remainder of cents left
    coins += cents / 10;
    cents %= 10;

    // Calculate how many nickels you need, and remainder of cents left
    coins += cents / 5;
    cents %= 5;

    // Rest of coins would be divided by one so can just count them
    coins += cents;

    // Output amount of total coins needed
    printf("%d\n", coins);

    return 0;
}
